src/cluster.c

```diff
@@ -6050,7 +6050,8 @@ int clusterRedirectBlockedClientIfNeeded(client *c) {
     if (c->flags & CLIENT_BLOCKED &&
         (c->btype == BLOCKED_LIST ||
          c->btype == BLOCKED_ZSET ||
-         c->btype == BLOCKED_STREAM))
+         c->btype == BLOCKED_STREAM ||
+         c->btype == BLOCKED_MODULE))
```

Comment:
need check `moduleClientIsBlockedOnKeys`, we discussed it in #9284

Reply:
yes the dictEntry judgement can work, but it's an implicit condition, a clear and explicit judgement is better.

Reply:
The client which is blocked for other reasons may also need to be redirected when  the cluster is down. So we may also need to deal with these clients.
https://github.com/redis/redis/blob/157454ffa4388143f9d68753d0a94bc684010367/src/cluster.c#L6058-L6065
Do you think so?